src/Raven.Server/Documents/ETL/EtlLoader.cs

```diff
@@ -670,44 +670,37 @@ public void HandleDatabaseRecordChange(DatabaseRecord record)
                 }
             }
 
-            Parallel.ForEach(toRemove, x =>
-            {
-                foreach (var process in x.Value)
-                {
-                    _database.DatabaseShutdown.ThrowIfCancellationRequested();
-
-                    try
-                    {
-                        string reason = GetStopReason(process, myRavenEtl, mySqlEtl, myOlapEtl, myElasticSearchEtl, myQueueEtl, responsibleNodes);
-
-                        process.Stop(reason);
-                    }
-                    catch (Exception e)
-                    {
-                        if (Logger.IsInfoEnabled)
-                            Logger.Info($"Failed to stop ETL process {process.Name} on the database record change", e);
-                    }
-                }
-            });
-
             LoadProcesses(record, myRavenEtl, mySqlEtl, myOlapEtl, myElasticSearchEtl, myQueueEtl, toRemove.SelectMany(x => x.Value).ToList());
 
-            Parallel.ForEach(toRemove, x =>
+            if (toRemove.Count == 0)
+                return;
+
```

Comment:
I know it is in the original code as well... but where do we remove the processes from the `_processes`? do we need to?

Reply:
This is handled inside `LoadProcesses()`.